packages/create-gatsby/src/__tests__/run.ts

```diff
@@ -0,0 +1,152 @@
+import { reporter } from "../utils/reporter"
+import { run } from "../index"
+
+jest.mock(`../utils/parse-args`)
+jest.mock(`enquirer`, () => {
+  const OriginalEnquirer = jest.requireActual(`enquirer`)
+
+  class MockedEnquirer extends OriginalEnquirer {
+    constructor() {
+      super()
+      // Turns waiting for user input off and autofills with answers
+      this.options = { show: false, autofill: true }
+
+      // Mock answers
+      this.answers = {
+        // First prompt answer
+        name: `hello-world`,
+
+        // Main question set answers
+        project: `hello-world`,
+        language: `js`,
+        cms: `none`,
+        styling: `none`,
+        features: [],
+
+        // Confirmation prompt answer
+        confirm: true,
+      }
+    }
+  }
+  return MockedEnquirer
+})
+jest.mock(`../utils/reporter`)
+jest.mock(`../tracking`, () => {
+  return {
+    trackCli: jest.fn(),
+  }
+})
+jest.mock(`../init-starter`, () => {
+  return {
+    initStarter: jest.fn(),
+    getPackageManager: jest.fn(),
+    gitSetup: jest.fn(),
+  }
+})
+jest.mock(`../install-plugins`, () => {
+  return {
+    installPlugins: jest.fn(),
+  }
+})
+jest.mock(`../utils/site-metadata`, () => {
+  return {
+    setSiteMetadata: jest.fn(),
+  }
+})
+jest.mock(`../utils/question-helpers`, () => {
+  const originalQuestionHelpers = jest.requireActual(
+    `../utils/question-helpers`
+  )
+  return {
+    ...originalQuestionHelpers,
+    validateProjectName: jest.fn(() => true),
+  }
+})
+jest.mock(`../components/utils`, () => {
+  return {
+    center: jest.fn(args => args),
+    wrap: jest.fn(args => args),
+  }
+})
+
+const dirName = `hello-world`
+let parseArgsMock
+
+describe(`run`, () => {
```

Comment:
Should have a test for the `ts` flag maybe? We could check this by testing the `trackCli`

Reply:
Good idea, added that and also checked by testing `initStarter`